model/pdata/common.go

```diff
@@ -525,6 +532,26 @@ func (am AttributeMap) Delete(key string) bool {
 	return false
 }
 
+// RemoveIf removes the entries for which the function in question returns true
+func (am AttributeMap) RemoveIf(f func(string, AttributeValue) bool) {
+	newLen := 0
+	for i := 0; i < len(*am.orig); i++ {
+		akv := &(*am.orig)[i]
+		if f(akv.Key, AttributeValue{&akv.Value}) {
+			continue
+		}
+		if newLen == i {
+			// Nothing to move, element is at the right place.
+			newLen++
+			continue
+		}
+		(*am.orig)[newLen] = (*am.orig)[i]
+		newLen++
+	}
+	// TODO: Prevent memory leak by erasing truncated values.
```

Comment:
Just curious: is it necessary to erase the individual elements? Is it not sufficient that the slice length is reduced? Doesn't GC understand that?

Reply:
I think you might be Right.  See my comment below

Reply:
removed the TODO